userapi/consumers/roomserver.go

```diff
@@ -417,6 +418,11 @@ func (s *OutputRoomEventConsumer) localRoomMembers(ctx context.Context, roomID s
 		return nil, 0, err
 	}
 
+	totalCount, err := s.rsAPI.JoinedUserCount(ctx, roomID)
```

Comment:
Right, I _think_ I understand what you're doing here. This needs more comments. It is my understanding that previously we:
 - Called `QueryMembershipsForRoom` to get ALL joined members in the room.
 - Therefore could use `len(res.JoinEvents)` for the joined count.

But now we:
- Call `QueryMembershipsForRoom` to get LOCAL joined members only.
- Need to call this `JoinedUserCount` to get the total joined count incl. remote users?

If so, my questions are:
- Presumably this optimisation does nothing if there are no remote users, or if the bulk of the members are local to this server.
- Is there any difference in perf getting local vs remote users? Presumably no, but because we often want to do operations on local users, there's no point pulling out remote users too?

Reply:
Yep, that's about it.
Previously we would parse the complete membership events for ALL joined users (which would `json.Unmarshal` for every event), now only for LOCAL users, which can be significantly lower than ALL users. (e.g. on my instance, there is only me in #dendrite, but 2k users over all - so instead of unmarshalling 2k times, we now do it once)
Once we unmarshalled the events, we also store it in our cache, which we may not even need there, so this possibly "pollutes" our cache with events we rarely care about.

If the majority of users is local, this doesn't do anything less than before, expect for one more DB hit to get the user count.